src/exit_node.hpp

```diff
@@ -34,12 +34,14 @@ template <typename ResponseType>
 class ExitNode : public Node {
     ResponseType* response;
     const tensor_map_t outputsInfo;
+    bool sharedInputContentsUsed;
 
 public:
-    ExitNode(ResponseType* response, const tensor_map_t& outputsInfo, std::set<std::string> gatherFromNode = {}) :
+    ExitNode(ResponseType* response, const tensor_map_t& outputsInfo, std::set<std::string> gatherFromNode = {}, bool sharedInputContentsUsed = true) :
```

Comment:
Minor - here we should rename this arg to:
useSharedInputContent -> we tell deserialization where to put the buffer